### PR TITLE
Adds "anchor" option

### LINKED ISSUE #38: Feature request: option to anchor the start of the track at the top of the donut
Is anyone working on adding an option to allow for anchoring the track at the top of the donut rather than the bottom?

### PR DESCRIPTION
**Pull request by tobie**
This option allows starting the donut from the top or the bottom. Defaults to bottom.

Closes #38.

### CODE REVIEW

File: src/donutty.js [hunk 1]
```diff
@@ -63,6 +63,7 @@
         this.options.title =       isDefined( this.options.title ) ? this.options.title : function() { return "Donut Chart Graphic"; };
         this.options.desc =        isDefined( this.options.desc ) ? this.options.desc : function( v ) { return "A donut chart ranging from " + v.min + " to " + v.max + " with a current value of " + v.value + "."; };
         this.options.dir =         isDefined( this.options.dir ) ? this.options.dir : false;
+        this.options.anchor =      (isDefined( this.options.anchor ) && this.options.anchor == "top") ? "top" : "bottom";
```
Comment (simeydotme) [inline src/donutty.js:66]: Nice PR!
I only ask if we can make these small tweaks so it matches the formatting and strict `===`

```suggestion
        this.options.anchor =      isDefined( this.options.anchor ) && this.options.anchor === "top" ? "top" : "bottom";
```

File: src/donutty.js [hunk 2]
```diff
@@ -196,7 +197,8 @@
 
         var viewbox = this.options.radius * 2 + this.options.thickness + 1,
             rotateExtra = this.options.round ? this.options.thickness / 3 : 0,
-            rotate = this.options.circle ? 90 + rotateExtra : -225,
+            rotateAnchor = this.options.anchor == "top" ? 180 : 0,
```
Comment (simeydotme) [inline src/donutty.js:200]:
```suggestion
            rotateAnchor = this.options.anchor === "top" ? 180 : 0,
```

**Review by simeydotme — CHANGES_REQUESTED** [review]
I've marked two things to change syntactically to align with the formatting!
Other than those, great PR!

P.S. Have you tested this with both a `circle` and non-`circle` donut chart? looks like it should just invert the bitten donut into kind of a horse-shoe? But I can double-check after the PR is ready!

**Comment by tobie** [comment]
> P.S. Have you tested this with both a circle and non-circle donut chart? looks like it should just invert the bitten donut into kind of a horse-shoe? But I can double-check after the PR is ready!

Yes, I have. It looks like an upside down gauge. Maybe that's weird? If so, we could make it for the full circle version only.

**Comment by simeydotme** [comment]
> > P.S. Have you tested this with both a circle and non-circle donut chart? looks like it should just invert the bitten donut into kind of a horse-shoe? But I can double-check after the PR is ready!
> 
> Yes, I have. It looks like an upside down gauge. Maybe that's weird? If so, we could make it for the full circle version only.

Oh cool, no no, not bothered if it's a little strange, the implementer can decide that. Just wanted to confirm its not doing anything unexpected (like sideways) :)

**Comment by tobie** [comment]
Hey, I’m not sure if there’s anything else you want me to do, here. I’m also confused about the labels you added to the pull request as “good first issue” and “help needed” labels are generally applied to issues before anyone steps up to offer a solution, not to pull requests or issues opened in order to discuss a feature prior to writing a PR for it.

**Comment by tobie** [comment]
> Just wanted to confirm its not doing anything unexpected (like sideways) :)

It is not!

**Comment by simeydotme** [comment]
Hi @tobie !
Thanks for this, it's been updated to `2.4.0` now with the new changes.

One thing; I had to revert the 180d rotation you added in favour of the scaling I mentioned before. This is because of the half-donut, specifically when "RTL" text mode was on. The rotation was negating the direction of the donut. Not a show-stopper, but it was weird.

Here's a little screencapture of the correctly functioning version (2.4.0) with scale instead of rotate! :)

donutty-tests

**Review by simeydotme — COMMENTED** [review]
👍

**Comment by tobie** [comment]
Thanks a lot!

**Pull request merged**